src/libraries/Common/src/System/Net/Http/Http3/Frames/Http3Frame.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+
+#if KESTREL
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http3
+#else
+namespace System.Net.Http
+#endif
+{
+    internal static partial class Http3Frame
+    {
+        public static bool TryReadIntegerPair(ReadOnlySpan<byte> buffer, out long a, out long b, out int bytesRead)
+        {
+            if (VariableLengthIntegerHelper.TryRead(buffer, out a, out int aLength))
+            {
+                buffer = buffer.Slice(aLength);
+                if (VariableLengthIntegerHelper.TryRead(buffer, out b, out int bLength))
+                {
+                    bytesRead = aLength + bLength;
+                    return true;
+                }
+            }
+
+            b = 0;
+            bytesRead = 0;
+            return false;
+        }
+
+        public static bool TryWriteFrameEnvelope(Http3FrameType frameType, long payloadLength, Span<byte> buffer, out int bytesWritten)
+        {
+            Debug.Assert(VariableLengthIntegerHelper.GetByteCount((long)frameType) == 1, $"{nameof(TryWriteFrameEnvelope)} assumes {nameof(frameType)} will fit within a single byte varint.");
```

Comment:
Should this be fixed to support multiple byte varints?

Reply:
It's an optimization. If there ever comes a time we have frame types over 1 byte we can update it.